src/jit/valuenum.cpp

```diff
@@ -6892,6 +6903,15 @@ void Compiler::fgValueNumberTree(GenTreePtr tree, bool evalAsgLhsInd)
                             vnStore->VNExcSetSingleton(vnStore->VNForFunc(TYP_REF, VNF_OverflowExc));
                         excSet = vnStore->VNPExcSetUnion(excSet, ValueNumPair(overflowExcSet, overflowExcSet));
                     }
+
+                    // Divides add a divide-by-zero exception.
+                    if (tree->OperIs(GT_DIV, GT_UDIV))
+                    {
+                        ValueNum divideByZeroExcSet =
+                            vnStore->VNExcSetSingleton(vnStore->VNForFunc(TYP_REF, VNF_DivideByZeroExc));
```

Comment:
> `VNF_DivideByZeroExc` needs to be updated to take the divisor as an argument

Yes. I was planning to address this in a follow-up change. Would you prefer that I do so with this change instead?

 > For that matter, it would be good to add the repro cases from #8648.

Good point.